libmscore/measurenumber.cpp

```diff
@@ -60,18 +122,80 @@ void MeasureNumber::layout()
       setPos(QPointF());
       if (!parent())
             setOffset(0.0, 0.0);
-      //else if (isStyled(Pid::OFFSET))
-      //      setOffset(propertyDefault(Pid::OFFSET).toPointF());
 
-      const StaffType* st = staff() ? staff()->constStaffType(measure()->tick()) : nullptr;
-      if (st && st->lines() == 1)
-            rypos() = (placeBelow() ? 2.0 : -2.0) * spatium();
+      // TextBase::layout1() needs to be called even if there's no measure attached to it.
+      // This happens for example in the palettes.
+      TextBase::layout1();
+      // this could be if (!measure()) but it is the same as current and slower
+      // See implementation of MeasureNumber::measure().
+      if (!parent())
```

Comment:
I think I need a comment, why it might not be parent, that this is a normal situation in this case, and not a programming error.

Reply:
isn't it already in the comment just above ? it says:
```
      // TextBase::layout1() needs to be called even if there's no measure attached to it.
      // This happens for example in the palettes.
```

It happens in the palettes

Reply:
I'm sorry, did not read carefully